src/Reflection/Annotations/AnnotationsPropertiesClassReflectionExtension.php

```diff
@@ -39,10 +40,20 @@ private function findClassReflectionWithProperty(
 	{
 		$propertyTags = $classReflection->getPropertyTags();
 		if (isset($propertyTags[$propertyName])) {
+			$propertyTag = $propertyTags[$propertyName];
+			$defaultType = $propertyTag->getReadableType() ?? $propertyTag->getWritableType();
+			if ($defaultType === null) {
+				throw new ShouldNotHappenException();
+			}
+
 			return new AnnotationPropertyReflection(
 				$declaringClass,
 				TemplateTypeHelper::resolveTemplateTypes(
-					$propertyTags[$propertyName]->getType(),
+					$propertyTag->getReadableType() ?? $defaultType,
```

Comment:
Another point of difference from #2294, this will continue to return an incorrect readable type when the property is not readable. It would be more elegant if we used `VoidType` as the fallback type – then instead of checking if the property is readable/writeable and checking if a value can inhabit the type, we could simply check the value, since `void` is a bottom type that cannot be inhabited by any value.

Reply:
That's NeverType, not VoidType. Yeah, I didn't understand that change. Feel free to update that scenario to NeverType, thank you.

Reply:
Opened #2328.